### ISSUE TITLE
Explain in docs how to wait for new messages in service bus queue

### ISSUE THREAD

**Issue by Korijn** [Azure/azure-sdk-for-python#513]
I'd like to know how I can react to new messages appearing in a service bus queue using the python sdk. As far as I know, I can write `msg = bus_service.receive_queue_message('taskqueue')`, but does that block/poll/something or does it always return immediately? This should be clarified in the docs.

Ideally, I'd like to be able to pass a timeout and wait for that time span to see if a new message appears when there aren't any available immediately. Supporting that feature asynchronously would be even better! Something like:

```python
while True:
    msg = await bus_service.receive_queue_message('taskqueue', wait=500)  # blocks/defers execution for 500 seconds
    if msg:
        process(msg)
```

**Comment by huguesv** [comment]
Hi Korijn,

Reading from an azure service bus queue will automatically block for a certain timeout period, which can be passed in as a parameter.

FYI: It's the opposite for azure storage queues, which do not block and return immediately if there are no messages.

From https://github.com/Azure/azure-sdk-for-python/blob/master/azure-servicebus/azure/servicebus/servicebusservice.py#L937

```

    def receive_queue_message(self, queue_name, peek_lock=True, timeout=60):
        '''
        Receive a message from a queue for processing.

        queue_name:
            Name of the queue.
        peek_lock:
            Optional. True to retrieve and lock the message. False to read and
            delete the message. Default is True (lock).
        timeout:
            Optional. The timeout parameter is expressed in seconds.
        '''
```

As you pointed out, the docs on azure.com/readthedocs.org don't mention that though, so we should update them.

**Comment by Korijn** [comment]
Ah, excellent! Thanks for the explanation. I browsed the code some more and have one more question: can you "renew" a lock? E.g. when processing will take more time than the lock duration allows for.

**Comment by huguesv** [comment]
https://msdn.microsoft.com/en-us/library/azure/jj839746.aspx

It's in the REST API but not exposed from Python.  It was probably added at a later time and we didn't notice we were missing it until now.

It should be relatively easy to add support for it.  If you'd like to give it a try, we're happy to take contributions!

**Comment by huguesv** [comment]
FYI: There is a max limit for the timeout of 60 secs (which is the default value).  If you try to use a larger value, you'll notice that it still returns after 60 secs.

**Comment by Korijn** [comment]
Ah, I see. Well, if 60 seconds is the default, we'll definitely be sending a PR for that feature pretty soon, since the processing will be done by a human. :)

**Comment by lmazuel** [comment]
The package 'azure-servicebus' was released yesterday as 0.20.3 with the renew lock feature.
I think it closes this issue, please comment on it if necesseray.
Thank you!
